k8s/k8s.go

```diff
@@ -103,6 +103,71 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 			logger.WithField("stdin", args.StdinData).Debug("Updated stdin data")
 		}
 
+		// Only used by K8s so if its null then we're not doing k8s stuff
+		var labels map[string]string
+		var annot map[string]string
+
+		// Only attempt to fetch the labels and annotations from Kubernetes
+		// if the policy type has been set to "k8s". This allows users to
+		// run the plugin under Kubernetes without needing it to access the
+		// Kubernetes API
+		if conf.Policy.PolicyType == "k8s" {
+			var err error
+
+			labels, annot, err = getK8sLabelsAnnotations(client, k8sArgs)
+			if err != nil {
+				// Cleanup IP allocation and return the error.
+				//	utils.ReleaseIPAllocation(logger, conf.IPAM.Type, args.StdinData)
+				return nil, err
+			}
+			logger.WithField("labels", labels).Debug("Fetched K8s labels")
+			logger.WithField("annotations", annot).Debug("Fetched K8s annotations")
+
+			// Add calico IPAM specific annotations only when the IPAM type is "calico-ipam"
+			if conf.IPAM.Type == "calico-ipam" {
+
+				v4pools := annot["cni.projectcalico.org/ipv4pools"]
+				v6pools := annot["cni.projectcalico.org/ipv6pools"]
+
+				if len(v4pools) != 0 || len(v6pools) != 0 {
+					var stdinData map[string]interface{}
+					if err := json.Unmarshal(args.StdinData, &stdinData); err != nil {
+						//	utils.ReleaseIPAllocation(logger, conf.IPAM.Type, args.StdinData)
+						return nil, err
+					}
+					var v4PoolSlice, v6PoolSlice []string
+
+					if len(v4pools) > 0 {
+						if err := json.Unmarshal([]byte(v4pools), &v4PoolSlice); err != nil {
+							//	utils.ReleaseIPAllocation(logger, conf.IPAM.Type, args.StdinData)
+							logger.WithField("IPv4Pool", v4pools).Error("Error parsing IPv4 IPPools")
+							return nil, err
+						}
+						stdinData["ipam"].(map[string]interface{})["ipv4_pools"] = v4PoolSlice
```

Comment:
Having an error check is also what I thought in the beginning, but the type of `stdinData` is always  `var stdinData map[string]interface{}`. I can add it, but not sure it's needed since type assertion for that type is always going to be true.